contracts/src/TeleporterTokenSource.sol

```diff
@@ -502,12 +521,14 @@ abstract contract TeleporterTokenSource is
         } else if (bridgeMessage.messageType == BridgeMessageType.REGISTER_DESTINATION) {
             RegisterDestinationMessage memory payload =
                 abi.decode(bridgeMessage.payload, (RegisterDestinationMessage));
+            require(
+                payload.sourceTokenDecimals == tokenDecimals, "TeleporterTokenSource: invalide token decmials"
+            );
```

Comment:
fixed the typo in https://github.com/ava-labs/teleporter-token-bridge/pull/142/commits/cd169d59033d4a1a573a7f04c3a7749c45458bd7

but i wanted to double check on the other part of this comment. are you sure that check shouldn't stay where it is? the information being checked here, `payload.sourceTokenDecimals`, isn't needed for anything in `_registerDestination`, so that's why I put the check outside of that function. and, i saw a precent for `require` statements in earlier branches of this `if`, so it didn't feel TOO inappropriate when i wrote it. but if you're sure that we should encapsulate the check within `_registerDestination` then I'm happy to do that, but i wanted to check first.

Reply:
Fair enough, definitely not a strong preference either way.

Reply:
I think the other require checks were added in this function because they were for `handleSendAndCall`, which needs to be implemented by derived contracts, and we wanted to avoid duplicating the code for the implementations, and to avoid from people forgetting to add those checks in their implementations.

In this case, there's just one `registerDesination` function so think we can move the logic there and just pass in the struct.